controllers/storageclient_controller.go

```diff
@@ -188,6 +192,15 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 		return res, err
 	}
 
+	storageClientResponse, err := externalClusterClient.GetStorageConfig(r.ctx, r.storageClient.Status.ConsumerID)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get StorageConfig: %v", err)
+	}
```

Comment:
A call to the provider to acquire additional desired state details should not be executed mid-reconcile.
Please move this line above to be executed after the onboarding part and before we starting to apply reconciliation logic to dependent resources (line 194)